frontend/simple/views/Invite.vue

```diff
@@ -164,21 +272,72 @@ export default {
             },
             mailbox
           )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
-
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          // We need to determine if the invitation needs a vote
```

Comment:
OK, here the logic needs to be separated out from the loop.

I understand that you seem to want this one widget to handle two very different cases (invitation at group creation, and as a proposal creation widget after group creation), and while that might be OK to do, it still doesn't make sense for the proposal part to be within a loop of adding members.

If you're at the point where you're proposing to add a member, you are not in a loop. The user is proposing to invite *one* member to the group, so obviously there shouldn't be a loop involved anywhere in that scenario.

Either:

- Clearly separate out the proposal part of this code from the group-creation-loop part
- Or, create two completely separate components, one for invitation at group-start, and one for proposing to invite a single member

I personally recommend creating two separate components, it will make it a lot easier to deal with the code and read it, IMO.